### PR TITLE
Add filter guide

### PR DESCRIPTION
**Pull request by XaosSintez**

### CODE REVIEW

File: spec/swagger.yaml [hunk 1]
```diff
@@ -323,7 +343,7 @@ parameters:
     description: |
       The collection items filter requires a special format.
       Use "," for multiple allowed values.  Use ";" for multiple fields.
-      See the filter guide for more options and examples about this format.
+      See the [filter guide](https://rebilly.github.io/RebillyAPI/#section/Using-filter) for more options and examples about this format.
```
Comment (viceromag) [inline spec/swagger.yaml]: Unrelated to this particular PR, but please add this guidelines link to User API too - https://github.com/Rebilly/RebillyUserAPI

File: spec/swagger.yaml [hunk 2]
```diff
@@ -65,6 +65,26 @@ info:
...
+
+    - You can use multiple values using `,` as values separator: `?filter=firstName:John,Bob`.
+
+    - To negate the filter use `!`: `?filter=firstName:!John`. Note that you can negate multiple values like this: `?filter=firstName:!John,Bob`. This filter rule will exclude all Johns and Bobs from the response.
+
+    - You can use range filters like this: `?filter:amount:1..10`.
+
+    - You can use gte (greater than or equals) filter like this: `?filter:amount:1..`, or lte (less than or equals) than filter like this: `?filter:amount:..10`.
+
+    - You can create some [predefined values lists](https://rebilly.github.io/RebillyUserAPI/#tag/Lists) and use them in filter: `?filter:firstName:@yourListName`.
```
Comment (viceromag) [inline spec/swagger.yaml]: Maybe add a reference to `not in list` (!@listname) too?

File: spec/swagger.yaml [hunk 3]
```diff
@@ -65,6 +65,26 @@ info:
...
     ```
+
+    # Using filter
+    Rebilly provides collections filtering. You can use `?filter` param on collection to define which records should be shown in the response.
+
+    Here is filter format description:
+
+    - Fields and values in filter are separated with `:`: `?filter=firstName:John`.
+
+    - Fields in filter are separated with `;`: `?filter:firstName:John;lastName:Doe`.
```
Comment (vova07) [inline spec/swagger.yaml]: Typo, should be: `?filter=`

File: spec/swagger.yaml [hunk 4]
```diff
@@ -65,6 +65,26 @@ info:
...
+
+    - You can use multiple values using `,` as values separator: `?filter=firstName:John,Bob`.
+
+    - To negate the filter use `!`: `?filter=firstName:!John`. Note that you can negate multiple values like this: `?filter=firstName:!John,Bob`. This filter rule will exclude all Johns and Bobs from the response.
+
+    - You can use range filters like this: `?filter:amount:1..10`.
+
+    - You can use gte (greater than or equals) filter like this: `?filter:amount:1..`, or lte (less than or equals) than filter like this: `?filter:amount:..10`.
```
Comment (vova07) [inline spec/swagger.yaml]: Typo, should be: `?filter=`

File: spec/swagger.yaml (hunk 2 above)
Comment (vova07) [inline spec/swagger.yaml]: Typo, should be: `?filter=`

File: spec/swagger.yaml (hunk 1 above)
Comment (XaosSintez) [reply spec/swagger.yaml]: Yes, as soon as i will merge this one

**Pull request merged**